### PR TITLE
fix: WT-2045 - fixing smartcheckout routing

### PR DESCRIPTION
**Pull request by andrearampin**
# Summary

# Customer Impact

## Security

# Things worth calling out

# Before submitting the PR, please consider the following:

- [ ] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.

### CODE REVIEW

File: packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts [hunk 1]
```diff
@@ -13,14 +13,21 @@ export const onRampRoute = async (
...
   let hasAllowList = false;
   const onRampAllowList = await allowListCheckForOnRamp(config, availableRoutingOptions);
 
   onRampAllowList.forEach((token) => {
-    if (token.address === required.token?.address) {
+    if (!token.address) return;
+    if (!required.token) return;
+    if (
+      isNativeToken(required.token.address)
+      || token.address.toLowerCase() === required.token.address?.toLowerCase()
```
Comment (dreamoftrees) [inline packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts:29]: @imx-mikhala and I had a chat about token address comparisons, this is also a good example of where a util helper to compare token address as case insensitive check would be handy.

**Review by dreamoftrees — APPROVED** [review]
Looks good thanks @andrearampin 👍

File: packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts (hunk 1 above)
Comment (imx-mikhala) [reply packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts:29]: lets fix that as part of `WT-1957`

File: packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts [hunk 2]
```diff
@@ -13,14 +13,21 @@ export const onRampRoute = async (
   availableRoutingOptions: AvailableRoutingOptions,
   balanceRequirement: BalanceRequirement,
 ): Promise<OnRampFundingStep | undefined> => {
-  if (balanceRequirement.type !== ItemType.ERC20 && balanceRequirement.type !== ItemType.NATIVE) return undefined;
-  const { required, current, delta } = balanceRequirement;
+  // Only check for ERC20 and NATIVE
+  if (![ItemType.ERC20, ItemType.NATIVE].includes(balanceRequirement.type)) return undefined;
```
Comment (imx-mikhala) [inline packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts:17]: can you add unit test for this line

**Review by imx-mikhala — APPROVED** [review]

**Pull request merged**